src/BankAccount.java

```diff
@@ -0,0 +1,123 @@
+import components.map.Map1L;
+
+/**
+ * Put a short phrase describing the program here.
+ *
+ * @author Put your name here
+ *
+ */
+public final class BankAccount extends Map1L<String, Double> {
```

Comment:
I would not extend map when you're also using it as the representation.

Reply:
With that said, I otherwise like this design! Having separate accounts map to pairs is nice.